M/MLX_C/build_tarballs.jl

```diff
@@ -0,0 +1,62 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "MLX_C"
+version = v"0.1.0"
+
+sources = [
+    GitSource("https://github.com/ml-explore/mlx-c.git", "e8889ddf56b3bd734eb79d5e5b13586a39c01403"),
+    ArchiveSource("https://github.com/roblabla/MacOSX-SDKs/releases/download/macosx14.0/MacOSX14.0.sdk.tar.xz",
+                  "4a31565fd2644d1aec23da3829977f83632a20985561a2038e198681e7e7bf49"),
+    DirectorySource("./bundled"),
+]
+
+script = raw"""
+if [[ "$target" == *-apple-darwin* ]]; then
+    apple_sdk_root=$WORKSPACE/srcdir/MacOSX14.0.sdk
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" $CMAKE_TARGET_TOOLCHAIN
+    sed -i "s!/opt/$bb_target/$bb_target/sys-root!$apple_sdk_root!" /opt/bin/$bb_full_target/$target-clang++
+
+    export MACOSX_DEPLOYMENT_TARGET=13.3
```

Comment:
Why using SDK 14.0 but saying that minimum supported target is 13.3?  Why not using https://github.com/roblabla/MacOSX-SDKs/releases/tag/13.3 instead?

Reply:
Using a newer SDK should be fine, you can build for older deployment targets on newer SDKs.

Also, it appears there is a mismatch between the deployment target here and the one in MLX (that this depends on):
```
[07:36:47] ld64.lld: warning: /workspace/destdir/lib/libmlx.dylib has version 13.5.0, which is newer than target minimum of 13.3.0
```
Since MLX is using the Python wheel on MacOS, the setting of the compatibility target isn't used at all in that build, so we need to follow whatever the minimum target in the Python wheel is.

Reply:
I have been using SDK 14.0 in both MLX and this due to this is the version required by MLX upstream (even though it's likely only necessary for the Metal backend which we are not building): https://github.com/search?q=repo%3Aml-explore%2Fmlx%20sdk&type=code

Reply:
Changed macOS target for aarch64-apple-darwin to 13.5